### PR TITLE
Assign random reviewer to PR

### PR DESCRIPTION
**Pull request by Voark**
## Summary
Assigns a random reviewer from `.github/reviewer-lottery.yaml` to the PR when opened. Does not assign to draft PRs.

### Why `pull_request` and `pull_request_target`? 
I needed to use `pull_request` to trigger the action to test it in this PR. Otherwise `pull_request_target` should theoretically work if a PR is made from a branch outside the repo so a reviewer would be assigned to an outside contributor.

## Test Plan
Should work $^{TM}$

## Issues

Closes #

### CODE REVIEW

**Comment by Voark** [comment]
Least messy actions PR history

File: .github/reviewer-lottery.yml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+groups:
+  - name: devs
+    reviewers: 1
+    usernames:
```
Comment (EricPedley) [inline .github/reviewer-lottery.yml:4]: You forgot to add @JacE070

**Review by EricPedley — CHANGES_REQUESTED** [review]

File: .github/reviewer-lottery.yml (hunk 1 above)
Comment (Voark) [reply .github/reviewer-lottery.yml:4]: Done

**Review by EricPedley — APPROVED** [review]
🚀

**Pull request merged**